ext/ffi/lib.rs

```diff
@@ -2071,32 +2036,59 @@ unsafe extern "C" fn noop_deleter_callback(
 }
 
 #[op(v8)]
-fn op_ffi_get_buf<FP, 'scope>(
+fn op_ffi_get_ptr_value<'scope>(
+  scope: &mut v8::HandleScope<'scope>,
+  state: &mut deno_core::OpState,
+  ptr: serde_v8::Value<'scope>,
+) -> Result<serde_v8::Value<'scope>, AnyError> {
+  check_unstable(state, "Deno.UnsafePointer#getPointerValue");
+
+  let ptr = if let Ok(value) = v8::Local::<v8::External>::try_from(ptr.v8_value)
+  {
+    value.value() as *const c_char
+  } else {
+    return Err(type_error("Invalid FFI pointer value"));
+  };
+
+  let ptr_value = ptr as usize;
+
+  let value: v8::Local<v8::Value> = if ptr as isize > MAX_SAFE_INTEGER {
+    v8::BigInt::new_from_u64(scope, ptr_value as u64).into()
+  } else {
+    v8::Number::new(scope, ptr_value as f64).into()
+  };
+
+  Ok(value.into())
+}
+
+#[op(v8)]
+fn op_ffi_get_buf<'scope>(
   scope: &mut v8::HandleScope<'scope>,
   state: &mut deno_core::OpState,
   src: serde_v8::Value<'scope>,
+  offset: usize,
   len: usize,
-) -> Result<serde_v8::Value<'scope>, AnyError>
-where
-  FP: FfiPermissions + 'static,
-{
+) -> Result<serde_v8::Value<'scope>, AnyError> {
   check_unstable(state, "Deno.UnsafePointerView#arrayBuffer");
 
-  let permissions = state.borrow_mut::<FP>();
-  permissions.check(None)?;
-
-  let ptr = if let Ok(value) = v8::Local::<v8::Number>::try_from(src.v8_value) {
-    value.value() as usize as *mut c_void
-  } else if let Ok(value) = v8::Local::<v8::BigInt>::try_from(src.v8_value) {
-    value.u64_value().0 as usize as *mut c_void
+  let ptr = if let Ok(value) = v8::Local::<v8::External>::try_from(src.v8_value)
+  {
+    value.value()
   } else {
-    return Err(type_error("Invalid FFI pointer value, expected BigInt"));
+    return Err(type_error("Invalid FFI pointer value"));
   };
 
   if std::ptr::eq(ptr, std::ptr::null()) {
     return Err(type_error("Invalid FFI pointer value, got nullptr"));
   }
 
+  let ptr = if offset != 0 {
+    // SAFETY: Offset is user provided
+    unsafe { ptr.add(offset) }
+  } else {
+    ptr
+  };
```

Comment:
Adding zero has no effect on the value?
```suggestion
// SAFETY: Offset is user provided
let ptr = unsafe {
  ptr.add(offset)
};
```

Reply:
Good point. I'll do this if we move forward with this PR.